File: api/src/main/java/com/android/api/contacts/AuthApi.java
package com.android.api.contacts;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class AuthApi {

    private String timeStamp;
    private String md5Key;

    private static final String KEY_PRIVATE = "ff28c030a8a73b3f90bdbbaef598657e86ea4ba4";
    private static final String KEY_PUBLIC = "2b67c95d9c57aee27f2b5ff6b525f681";

    public AuthApi() {
        timeStamp = String.valueOf(System.currentTimeMillis());
    }

    public String getPublicKey() {
        return KEY_PUBLIC;
    }

    public String getPrivateKey() {
        return KEY_PRIVATE;
    }

    public String getTimeStamp() {
        return timeStamp;
    }

    public String getMd5Key() {

        String input = getTimeStamp() + getPrivateKey() + getPublicKey();

        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] md5Bytes = md.digest(input.getBytes());

            StringBuilder md5 = new StringBuilder();
            for (byte md5Byte : md5Bytes) {
                md5.append(Integer.toHexString((md5Byte & 0xFF) | 0x100).substring(1, 3));
            }
            md5Key = md5.toString();

        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return md5Key;
    }

}

File: api/src/main/java/com/android/api/AuthApi.kt
package com.android.api

import java.math.BigInteger
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

class AuthApi {

    companion object {
        private const val KEY_PRIVATE = "ff28c030a8a73b3f90bdbbaef598657e86ea4ba4"
        private const val KEY_PUBLIC = "2b67c95d9c57aee27f2b5ff6b525f681"
    }

    val timeStamp: String
        get() = System.currentTimeMillis().toString()

    val publicKey: String
        get() = KEY_PUBLIC

    val privateKey: String
        get() = KEY_PRIVATE

    var md5Key: String? = null
        get() {
            val input = timeStamp + KEY_PRIVATE + KEY_PUBLIC

            return try {
                val md = MessageDigest.getInstance("MD5")
                val messageDigest = md.digest(input.toByteArray())
                val number = BigInteger(1, messageDigest)
                var md5: String = number.toString(16)
                while (md5.length < 32) md5 = "0$md5"
                md5
            } catch (e: NoSuchAlgorithmException) {
                null
            }
        }
        private set

}